### DISCUSSION TITLE
Kornia Contrast Enhancement vs OpenCV

### DISCUSSION THREAD

**Discussion by rkalghatgi** [kornia/kornia#1659] [Questions]
How does Kornia's implementation of contrast enhancement (hist eq, CLAHE) compare in terms of performance to those available in OpenCV? Thank you.

**Comment by edgarriba** [comment]
@rkalghatgi we don't have any official benchmark on that functionality. But I believe @lferraz have some intuitions that got during the development.

**Reply by lferraz** [reply]
The code is quite optimized for GPUs and returns, as far as I remember, almost the same result as OpenCV. Regarding the speed it depends on several factors. If you want to use the CPU version I'm pretty sure that OpenCV is faster. If you want to use a GPU device the Kornia ones should be faster but of course depends on the GPU model and the location (RAM or VRAM) of the images.

My recomendation, check both on your hardware, it is quite straightforward to run this comparison. As far as I remember they have the exact same parameters.
